### PR TITLE
Remove ADResidual and its ilk

### LINKED ISSUE #13599: Too many AD types
## Bug Description
We have too many AD types. There's really no reason to have `ADResidual` for example. To be consistent with how all our traditional compute objects return `Real` for residuals and Jacobians, we should return `ADReal` from compute objects. I honestly thought that @dschwen and I had done this, but I'm sitting here in training and I'm seeing that it's everywhere 😢 

## Steps to Reproduce
Look in `MooseTypes`

## Impact
We are inundating our users with names that they have to remember. Additionally, someone coming from our old systems should have as consistent an interface as possible.

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #13599

### CODE REVIEW

**Comment by lindsayad** [comment]
Ok folks, don't merge! I can update all the apps to use the types we want them to use before this is merged, so we should not experience any period of breakage!

**Review by fdkong — APPROVED** [review]

**Pull request merged**
